core/src/main/java/org/hibernate/search/jsr352/internal/steps/lucene/EntityReader.java

```diff
@@ -242,7 +242,8 @@ private ScrollableResults buildScrollUsingHQL(StatelessSession ss, String HQL) {
 	private ScrollableResults buildScrollUsingCriteria(StatelessSession ss,
 			PartitionBound unit, Object checkpointId, JobContextData jobData) {
 
-		String idName = MassIndexerUtil.getIdName( entityType, session );
+		String idName = Projections.id().toString();
```

Comment:
Using `toString()` for anything else than debugging seems like a bad idea to me in general, unless an interface/class documents a specific behavior for this method. In this case, it seems that `Projection` does not give any special meaning to `toString()`. It seems the id projection's `toString()` method will just return "id".

Judging from the code in `Projections` and `org.hibernate.criterion.Order.toSqlString(Criteria, CriteriaQuery)`, I don't think this will work as soon as the id column is named something other than `id`. I'm not 100% sure, but this probably should be checked. Maybe you could test that?

From what I can see in `org.hibernate.criterion.Order.toSqlString(Criteria, CriteriaQuery)`, there is some support for aliases, though. So maybe you could just alias the id projection to something (say, `Projections.alias( Projections.id(), "aliased_id" )`) and order by the alias (`Order.property( "aliased_id" )`)?